repository/src/main/java/org/apache/atlas/repository/store/aliasstore/ESAliasStore.java

```diff
@@ -165,32 +170,46 @@ private Map<String, Object> getFilterForPurpose(AtlasEntity purpose) throws Atla
 
     private void personaPolicyToESDslClauses(List<AtlasEntity> policies,
                                              List<Map<String, Object>> allowClauseList) throws AtlasBaseException {
+        List<String> terms = new ArrayList<>();
         for (AtlasEntity policy: policies) {
 
             if (policy.getStatus() == null || AtlasEntity.Status.ACTIVE.equals(policy.getStatus())) {
                 List<String> assets = getPolicyAssets(policy);
 
                 if (getIsAllowPolicy(policy)) {
                     if (getPolicyActions(policy).contains(ACCESS_READ_PERSONA_METADATA)) {
+
+                        if (terms.size() + assets.size() + 1 > assetsMaxLimit) {
```

Comment:
nit: There should be a comment here clarifying the "why" behind this clause (why +1, etc.).

Reply:
Makes sense, I will do that